src/org/omegat/core/CoreEvents.java

```diff
@@ -111,9 +111,12 @@ public static void registerEditorEventListener(final IEditorEventListener listen
     public static void unregisterEditorEventListener(final IEditorEventListener listener) {
         EDITOR_EVENT_LISTENERS.remove(listener);
     }
+    
+    public static boolean isInProjectEventLoop = false;
```

Comment:
The field should not expose as public but should be private and provide getter at least.
For thread safety, it should be `volative` at least.

Please consider avoiding status in the `CoreEvent` class.